api/tests/views_tests.py

```diff
@@ -201,3 +214,166 @@ def test_post_relayaddress_flagged_error(free_user, free_api_client) -> None:
         "detail": "Your account is on pause.",
         "error_code": "account_is_paused",
     }
+
+
+class TermsAcceptedUserViewTest(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.path = "/api/v1/terms-accepted-user/"
+        self.fxa_verify_path = authentication.INTROSPECT_TOKEN_URL
+        self.uid = "relay-user-fxa-uid"
+
+    def _setup_client(self, token):
+        self.client = APIClient()
+        self.client.credentials(HTTP_AUTHORIZATION=f"Bearer {token}")
+
+    def tearDown(self):
+        cache.clear()
+
+    def test_no_authorization_header_returns_400(self):
+        client = APIClient()
+        response = client.post(self.path)
+
+        assert response.status_code == 400
+        assert response.json()["detail"] == "Missing Bearer header."
+
+    def test_no_token_returns_400(self):
+        client = APIClient()
+        client.credentials(HTTP_AUTHORIZATION="Bearer ")
+        response = client.post(self.path)
+
+        assert response.status_code == 400
+        assert response.json()["detail"] == "Missing FXA Token after 'Bearer'."
+
+    @responses.activate()
+    def test_invalid_bearer_token_error_from_fxa_returns_500_and_cache_returns_500(
+        self,
+    ):
+        fxa_response = _setup_fxa_response(401, {"error": "401"})
+        not_found_token = "not-found-123"
+        self._setup_client(not_found_token)
+
+        assert cache.get(get_cache_key(not_found_token)) is None
+
+        response = self.client.post(self.path)
+        assert response.status_code == 500
+        assert response.json()["detail"] == "Did not receive a 200 response from FXA."
+        assert responses.assert_call_count(self.fxa_verify_path, 1) is True
+        assert cache.get(get_cache_key(not_found_token)) == fxa_response
+
+        # now check that the code does NOT make another fxa request
+        response = self.client.post(self.path)
+        assert response.status_code == 500
+        assert response.json()["detail"] == "Did not receive a 200 response from FXA."
+        assert responses.assert_call_count(self.fxa_verify_path, 1) is True
```

Comment:
suggestion (non-blocking): Looks like this block is repeated 5 times. Refactor into a method?

```python
def _assert_cached_response(self, status_code: int, detail: str):
    response = self.client.post(self.path)
    assert response.status_code == status_code
    assert response.json()["detail"] == detail
    assert responses.assert_call_count(self.fxa_verify_path, 1) is True
```

Though this function depends on the `responses` scope in addition to `self` ... so maybe it's not as clean?

Reply:
Adding this as a part of clean-up ticket created in [MPP-3162](https://mozilla-hub.atlassian.net/browse/MPP-3162).